### ISSUE TITLE
Unit test CondToolsLHCInfoNewPopConTest failure: written wrong number of payloads

### ISSUE THREAD

**Issue by iarspider** [cms-sw/cmssw#42828]
Unit test `CondTools/RunInfo/CondToolsLHCInfoNewPopConTest` is failing for all architectures since CMSSW_13_3_X_2023-09-18-1100:

```
===== Test "CondToolsLHCInfoNewPopConTest" ====
testing LHCInfoPerFillPopConAnalyzer in EndFill mode for startTime="2022-10-24 01:00:00.000" endTime="2022-10-24 20:00:00.000"
testing LHCInfoPerLSPopConAnalyzer in endFill mode for startTime="2022-10-24 01:00:00.000" endTime="2022-10-24 20:00:00.000"
testing LHCInfoPerLSPopConAnalyzer in endFill mode for startTime="2022-07-11 22:00:00.000" endTime="2022-07-12 18:10:10.000"
Failure LHCInfoPerLSPopConAnalyzer in endFill mode written wrong number of payloads: Expected 69, but got 70: status 1

---> test CondToolsLHCInfoNewPopConTest had ERRORS
TestTime:34
^^^^ End Test CondToolsLHCInfoNewPopConTest ^^^^
```

**Comment by iarspider** [comment]
assign db

**Comment by aandvalenzuela** [comment]
Hello,

I can reproduce the issue in IB `CMSSW_13_3_X_2023-09-18-2300`, but also in older IBs such as `CMSSW_13_3_X_2023-09-15-2300` (I was trying to get the "old" payloads to check from where the extra one comes from).

It fails in https://github.com/cms-sw/cmssw/blob/master/CondTools/RunInfo/test/test_lhcInfoNewPopCon.sh#L48, and the list of payloads is:
```
Singularity> cat ls_end_test2.log | grep -E '^Since '        
Since 1
Since 7119252977130405888
Since 7119255781744050176
Since 7119262589267214336
Since 7119264964384129024
Since 7119283084851150848
Since 7119283183635398656
Since 7119283286714613760
Since 7119283385498861568
Since 7119283484283109376
Since 7119283587362324480
Since 7119283686146572288
Since 7119283784930820096
Since 7119283888010035200
Since 7119283986794283008
Since 7119284085578530816
Since 7119284188657745920
Since 7119284287441993728
Since 7119284386226241536
Since 7119284725528657920
Since 7119284785658200064
Since 7119284888737415168
Since 7119284987521662976
Since 7119285086305910784
Since 7119285189385125888
Since 7119285288169373696
Since 7119285386953621504
Since 7119285490032836608
Since 7119285588817084416
Since 7119285687601332224
Since 7119285790680547328
Since 7119285889464795136
Since 7119285988249042944
Since 7119286091328258048
Since 7119286190112505856
Since 7119286288896753664
Since 7119286387681001472
Since 7119286490760216576
Since 7119286589544464384
Since 7119286688328712192
Since 7119286791407927296
Since 7119287392703348736
Since 7119287491487596544
Since 7119287590271844352
Since 7119287689056092160
Since 7119287792135307264
Since 7119287890919555072
Since 7119287989703802880
Since 7119288092783017984
Since 7119288191567265792
Since 7119288290351513600
Since 7119288393430728704
Since 7119288492214976512
Since 7119288586704257024
Since 7119288685488504832
Since 7119288784272752640
Since 7119288887351967744
Since 7119288986136215552
Since 7119289084920463360
Since 7119290588159016960
Since 7119291391317901312
Since 7119303619089793024
Since 7119308854654926848
Since 7119348398418821120
Since 7119360338427904000
Since 7119521429766275072
Since 7119539120736567296
Since 7119544128668434432
Since 7119548432225665024
Since 7119550004183695360
```
Looking at the log I see 68 payloads received:
```
%MSG-i transferPayloads:  LHCInfoPerLSPopConAnalyzer:Test1@endJob  20-Sep-2023 10:47:30 CEST EndJob
TRANSFERED COND IOVS: 7119252977130405888 7119255781744050176 7119262589267214336 7119264964384129024 
7119283084851150848 7119283183635398656 7119283286714613760 7119283385498861568 7119283484283109376 
7119283587362324480 7119283686146572288 7119283784930820096 7119283888010035200 7119283986794283008 
7119284085578530816 7119284188657745920 7119284287441993728 7119284386226241536 7119284725528657920 
7119284785658200064 7119284888737415168 7119284987521662976 7119285086305910784 7119285189385125888 
7119285288169373696 7119285386953621504 7119285490032836608 7119285588817084416 7119285687601332224 
7119285790680547328 7119285889464795136 7119285988249042944 7119286091328258048 7119286190112505856 
7119286288896753664 7119286387681001472 7119286490760216576 7119286589544464384 7119286688328712192 
7119286791407927296 7119287392703348736 7119287491487596544 7119287590271844352 7119287689056092160 
7119287792135307264 7119287890919555072 7119287989703802880 7119288092783017984 7119288191567265792 
7119288290351513600 7119288393430728704 7119288492214976512 7119288586704257024 7119288685488504832 
7119288784272752640 7119288887351967744 7119288986136215552 7119289084920463360 7119290588159016960 
7119291391317901312 7119303619089793024 7119308854654926848 7119348398418821120 7119360338427904000 
7119521429766275072 7119539120736567296 7119544128668434432 7119548432225665024
%MSG
```
And 2 empty ones generated:
```
%MSG-i LHCInfoPerLSPopConSourceHandler:   LHCInfoPerLSPopConAnalyzer:Test1@endJob 20-Sep-2023 10:47:28 CEST EndJob
Added empty payload with IOV1 ( 1970-01-01T00:00:00 )

%MSG-i LHCInfoPerLSPopConSourceHandler:   LHCInfoPerLSPopConAnalyzer:Test1@endJob 20-Sep-2023 10:47:30 CEST EndJob
Added empty payload with IOV7119550004183695360 ( 2022-07-12T18:10:10 )
```

**Comment by perrotta** [comment]
+db
- Fixed (also in IB) with [#42837](https://github.com/cms-sw/cmssw/pull/42837)
